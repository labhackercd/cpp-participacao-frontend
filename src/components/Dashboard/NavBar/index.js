import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles, withStyles } from '@material-ui/core/styles';
import {AppBar, Tabs, Tab, Typography, Box, Grid, IconButton} from '@material-ui/core';
import { ReactComponent as Logo }from './../../../assets/logo.svg';
import { ReactComponent as LogoutIcon } from './../../../assets/user_logout_icon.svg';

const CustomTab = withStyles({
  root: {
    textTransform: 'capitalize',
  }
})(Tab);

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    backgroundColor: theme.palette.background.paper,
  },
  tab: {
    height: '100%',
    textTransform: 'capitalize',
  },
  logo: {
    marginLeft: theme.spacing(2),
  }
}));

function a11yProps(index) {
  return {
    id: `simple-tab-${index}`,
    'aria-controls': `simple-tabpanel-${index}`,
  };
}

export default function NavBar() {
  const classes = useStyles();
  const [value, setValue] = React.useState(0);

  const handleChange = (event, newValue) => {
   setValue(newValue);
 };

  return (
     <div className={classes.root}>
       <AppBar position="static">
       <Grid container>
          <Grid item md={2}>
            <div className={classes.logo}>
              <IconButton>
                <Logo />
              </IconButton>
            </div>
          </Grid>
          <Grid item md={9}>
            <Tabs
                value={value}
                onChange={handleChange}
                TabIndicatorProps={{style: {background: '#00C354'}}}
                className={classes.tab}>
              <CustomTab id="tabs-1" label="Dashboard" {...a11yProps(0)} />
              <CustomTab id="tabs-2" label="Configurações" {...a11yProps(1)} />
              <CustomTab id="tabs-3" label="Documentação" {...a11yProps(2)} />
            </Tabs>
          </Grid>
          <Grid item md={1}>
            <div className={classes.logout}>
              <IconButton>
                <LogoutIcon />
              </IconButton>
            </div>
          </Grid>
       </Grid>
       </AppBar>
     </div>
   );
}
