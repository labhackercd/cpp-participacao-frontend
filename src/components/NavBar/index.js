import React, { useEffect } from 'react';
import { makeStyles, withStyles } from '@material-ui/core/styles';
import {AppBar, Tabs, Toolbar, Tab, Grid, IconButton} from '@material-ui/core';
import { ReactComponent as Logo }from './../../assets/logo.svg';
import { ReactComponent as LogoutIcon } from './../../assets/user_logout_icon.svg';
import DashboardInitial from './../DashboardInitial'
import LogoutMenu from '../LogoutMenu/index'


const CustomTab = withStyles({
  root: {
    textTransform: 'capitalize',
  }
})(Tab);

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.black.main,
    height: '5vh',
  },
  tab: {
    height: '92%',
    textTransform: 'capitalize',
    margin: '0.5rem 0 0 6rem',
  },
  menu:{
    "& .MuiPaper-root-16": {
      backgroundColor: theme.palette.mediumGrey.main,
    },
    '& div': {
      width: "14%"
    }
  },

}));

function a11yProps(index) {
  return {
    id: `simple-tab-${index}`,
    'aria-controls': `simple-tabpanel-${index}`,
  };
}


export default function NavBar(props) {
  const classes = useStyles();
  const [logoutMenuOpen, setLogoutMenuOpen] = React.useState(false);
  const [anchorEl, setAnchorEl] = React.useState()

  const recordButtonPosition = (event) => {
    setAnchorEl(event.currentTarget);
    setLogoutMenuOpen(true);
  }
  const handleLogoutMenuClose = () => {
    setLogoutMenuOpen(false);
  };

  useEffect(() => {
    
  },[props.value]);

  return (
  
       <AppBar position="static">
        <Toolbar className={classes.root}>
           <Grid container>
              <Grid item md={2}>
                <div className={classes.logo}>
                  <IconButton><Logo/></IconButton>
                </div>
              </Grid>
              <Grid item md={9}>
              {props.value != undefined &&
                <Tabs value={props.value} onChange={props.handleTabPanelChange} TabIndicatorProps={{style: {background: '#00C354'}}} className={classes.tab}>
                  <CustomTab  label="Dashboard" {...a11yProps(0)} />
                  <CustomTab  label="Configurações" {...a11yProps(1)} />
                  <CustomTab  label="Documentação" {...a11yProps(2)} />
                </Tabs>
                }
              </Grid>
              <Grid item md={1}>
                <div className={classes.logout}>
                  <IconButton id="logoutMenu" onClick={(e) => { recordButtonPosition(e) }} aria-haspopup="true"><LogoutIcon /></IconButton>
                  <LogoutMenu open={logoutMenuOpen} handleClose={handleLogoutMenuClose} classes={classes} anchorEl={anchorEl}></LogoutMenu>
                </div>
              </Grid>
           </Grid>
        </Toolbar>
       </AppBar>
   );
}
