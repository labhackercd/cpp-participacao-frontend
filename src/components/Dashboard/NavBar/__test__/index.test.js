import React from 'react';
import ReactDOM from 'react-dom';
import MockTheme from './../../../mockTheme';
import {shallow, render, mount} from "enzyme";
import NavBar from './../index';

it("snapshot should not have changes", () => {
    const component = shallow(<NavBar/>);
    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});

test('Test if NavBar renders without crash', () => {
    const div = document.createElement("div")
    ReactDOM.render(<NavBar></NavBar>, div)
    ReactDOM.unmountComponentAtNode(div)
});

test('Test if clicking on tab button doesnt break page', () => {
    const wrapper = mount(<NavBar />);
    const tabButton = wrapper.find("#simple-tab-0").last();
    tabButton.simulate('click');
});
