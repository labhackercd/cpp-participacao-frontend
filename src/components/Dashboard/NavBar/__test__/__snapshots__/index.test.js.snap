// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should not have changes 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={2}
        >
          <div
            className="makeStyles-logo-3"
          >
            <WithStyles(ForwardRef(IconButton))>
              <ForwardRef(SvgLogo) />
            </WithStyles(ForwardRef(IconButton))>
          </div>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={9}
        >
          <WithStyles(ForwardRef(Tabs))
            TabIndicatorProps={
              Object {
                "style": Object {
                  "background": "#00C354",
                },
              }
            }
            className="makeStyles-tab-2"
            onChange={[Function]}
            value={0}
          >
            <WithStyles(WithStyles(ForwardRef(Tab)))
              aria-controls="simple-tabpanel-0"
              id="simple-tab-0"
              label="Dashboard"
            />
            <WithStyles(WithStyles(ForwardRef(Tab)))
              aria-controls="simple-tabpanel-1"
              id="simple-tab-1"
              label="Configurações"
            />
            <WithStyles(WithStyles(ForwardRef(Tab)))
              aria-controls="simple-tabpanel-2"
              id="simple-tab-2"
              label="Documentação"
            />
          </WithStyles(ForwardRef(Tabs))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={1}
        >
          <div
            className="makeStyles-logout-4"
          >
            <WithStyles(ForwardRef(IconButton))>
              <ForwardRef(SvgUserLogoutIcon) />
            </WithStyles(ForwardRef(IconButton))>
          </div>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
